src/test/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/unit/service/ApplicationServiceTest.kt

```diff
@@ -1040,6 +1049,39 @@ class ApplicationServiceTest {
       assertThat(validatableActionResult.message).isEqualTo("This application has already been submitted")
     }
 
+    @Test
+    fun `updateApprovedPremisesApplication returns GeneralValidationError when application has AP type specified in multiple ways`() {
```

Comment:
i can see lots of good tests for when defining AP Type in multiple ways, but do we have any when defining apType enum instead of booleans?

Reply:
I've added tests for submitting and updating the application using the `ApType` enum.